src/foam/nanos/http/FileServlet.java

```diff
@@ -0,0 +1,63 @@
+package foam.nanos.http;
+
+import com.google.api.client.util.IOUtils;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+public class FileServlet
+    extends HttpServlet
+{
+
+  public static final String SERVLET_NAME = "file";
+
+  private void fileNotFoundError(HttpServletResponse resp, String file) {
+    resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
+    resp.setContentType("application/json");
+
+    try {
+      PrintWriter pw = resp.getWriter();
+      pw.write(
+          "{\"error\": \"File not found\"," +
+              "\"filename\": \""+file+"\"}");
+    } catch (IOException ignore) {}
+  }
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    String pathInfo = req.getPathInfo();
+    String filePath = pathInfo.substring(SERVLET_NAME.length() + 2);
+    String fileName = "";
+
+    String[] paths = filePath.split("/");
```

Comment:
If I insert "../" into the path, does it let me read from a parent/ancestor directory?